### DISCUSSION TITLE
publisher how to sent two or more topics to the subscriber?

### DISCUSSION THREAD

**Discussion by zixuan-su** [eProsima/Fast-DDS#2736] [Q&A] [answered]
hi, i am a freshman ,i learn fastdds for three days , i have passed that publisher sent one topic to the subscriber ,but when i want to try more, i failed, publisher can sent two topic, but subscriber only can receive a topic ,could  you please  give me some examples?
thanks.... i have searched many reference , but  cant to answer my question.

**Comment by EduPonz** [comment, marked as answer]
This is a duplicate of #2663. There, it is explained that to received data in two different topics, you'd need two different data readers (one on each topic). It is also explained how to get the data on the subscriber callbacks instead of the data readers'.

**Reply by zixuan-su** [reply]
thanks! i have seen it , now i am looking up how to use subscriber callbacks to receive two topics.
In "HelloWorld" example, it uses the function "void on_data_available(DataReader* reader)", I would instead of "on_data_on_readers() "?
Sorry, I am very confused, I want to search some codes, ..I will make more effort to try.
Thank you again for your help.

**Reply by EduPonz** [reply]
When [creating a Subscriber](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/subscriber/subscriber/createSubscriber.html#creating-a-subscriber), you can pass a [SubscriberListener](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/subscriber/subscriberListener/subscriberListener.html#subscriberlistener) to it. As explained in [Listener](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/core/entity/entity.html#listener), `SubscriberListener` inherits from `DataReaderListener`. Mind that when setting a `SubscriberListener`, `on_data_on_readers` will be called unless otherwise specified in the mask. That means that  `on_data_available` will not be called in that case.

**Reply by zixuan-su** [reply]
I'll take a closer look at your example, thanks for your reply!!!

**Reply by zixuan-su** [reply]
hi,Can you help check my code?[Pub](https://github.com/zixuan-su/fastdds_/blob/main/Publisher.cpp)
[Sub](https://github.com/zixuan-su/fastdds_/blob/main/Sunscriber.cpp)
i.............

**Reply by EduPonz** [reply]
Hi @zixuan-su ,

Unfortunately I don't have time at the moment to review your code. You could try to reach support@eprosima.com to get some commercial support on the matter.
